pycbc/waveform/bank.py

```diff
@@ -721,7 +728,7 @@ def __getitem__(self, index):
 
         # Get the waveform filter
         distance = 1.0 / DYN_RANGE_FAC
-        if self.compressed_waveforms is not None :
+        if self.check_compressed_waveforms :
```

Comment:
Again, replace this with `if self.has_compressed_waveforms and self.use_compressed:`.